auth-service/src/main/kotlin/uk/gov/communities/delta/auth/controllers/external/DeltaLoginController.kt

```diff
@@ -162,9 +164,11 @@ class DeltaLoginController(
                     )
                 }
 
-                val authCode = authenticationCodeService.generateAndStore(
-                    userCn = loginResult.user.cn, client = client, traceId = call.callId!!
-                )
+                val authCode = withContext(Dispatchers.IO) {
```

Comment:
Is it not possible to put the `withContext(Dispatchers.IO)` inside the `generateAndStore` function? Does it need done at the interface like this?

Reply:
Yep could, the reason I didn't do that is because the other function (lookupAndInvalidate) was already called from an IO thread. I don't think doing a nested `withContext(Dispatchers.IO)` is really an, I'll maybe try switching to that.